src/complex.jl

```diff
@@ -0,0 +1,288 @@
+
+See also [`conj`](@ref conj), [`real`](@ref), [`imag`](@ref).
+"""
+ordinary_variable(x::AbstractVariable) = x
+
+"""
+    conj(x::AbstractVariable)
+
+Return the complex conjugate of a given variable if it was declared as a complex variable; else return the
+variable unchanged.
+
+    conj(x::AbstractMonomial)
+    conj(x::AbstractVector{<:AbstractMonomial})
+    conj(x::AbstractTerm)
+    conj(x::AbstractPolynomial)
+
+Return the complex conjugate of `x` by applying conjugation to all coefficients and variables.
+
+See also [`iscomplex`](@ref iscomplex), [`isconj`](@ref isconj).
+"""
+Base.conj(x::AbstractVariable) = x
+
+"""
+    real(x::AbstractVariable)
+
+Return the real part of a given variable if it was declared as a complex variable; else return the variable
+unchanged.
+
+    real(x::AbstractMonomial)
+    real(x::AbstractVector{<:AbstractMonomial})
+    real(x::AbstractTerm)
+    real(x::AbstractPolynomial)
+
+Return the real part of `x` by applying `real` to all coefficients and variables; for this purpose, every complex-valued
+variable is decomposed into its real- and imaginary parts.
+
+See also [`iscomplex`](@ref iscomplex), [`isrealpart`](@ref isrealpart), [`imag`](@ref).
+"""
+Base.real(x::AbstractVariable) = x
+
+"""
+    imag(x::AbstractVariable)
+
+Return the imaginary part of a given variable if it was declared as a complex variable; else return zero.
+
+    imag(x::AbstractMonomial)
+    imag(x::AbstractVector{<:AbstractMonomial})
+    imag(x::AbstractTerm)
+    imag(x::AbstractPolynomial)
+
+Return the imaginary part of `x` by applying `imag` to all coefficients and variables; for this purpose, every complex-valued
+variable is decomposed into its real- and imaginary parts.
+
+See also [`iscomplex`](@ref iscomplex), [`isimagpart`](@ref isimagpart), [`real`](@ref).
+"""
+Base.imag(::AbstractVariable) = MA.Zero()
+
+# extend to higher-level elements. We make all those type-stable (but we need convert, as the construction method may
+# deliver simpler types than the inputs if they were deliberately casted, e.g., term to monomial)
+function iscomplex(p::AbstractPolynomialLike)
+    for v in variables(p)
+        if !iszero(maxdegree(p, v)) && iscomplex(v)
+            return true
+        end
+    end
+    return false
+end
+Base.conj(x::M) where {M<:AbstractMonomial} =
+    convert(M, reduce(*, conj(var)^exp for (var, exp) in zip(variables(x), exponents(x)); init=constantmonomial(x)))
+Base.conj(x::V) where {V<:AbstractVector{<:AbstractMonomial}} = monovec(conj.(x))
+Base.conj(x::T) where {T<:AbstractTerm} = convert(T, conj(coefficient(x)) * conj(monomial(x)))
+Base.conj(x::P) where {P<:AbstractPolynomial} = iszero(nterms(x)) ? x : convert(P, sum(conj(t) for t in x))
+
+# Real and imaginary parts are harder to realize. The real part of a monomial can easily be a polynomial.
+for fun in [:real, :imag]
+    eval(quote
+        function Base.$fun(x::Union{AbstractMonomial,AbstractTerm})
```

Comment:
```suggestion
        function Base.$fun(x::APL)
```

Reply:
Yes, this would be a way. Perhaps more memory-hungry over all, but faster.
My initial thoughts were (if I remember correctly) that if I do the substitions on a monomial-level first, this might already lead to some cancellations in the expanded monomials, and very likely to dropping a lot of terms (because only the real or imaginary part is retained). So I expand (= large number of terms) the small parts first, leading to simplifications on this level, then add them all up. If the same function is used on the APL level, everything is expanded first, then potentially dropped by the application of the function. Of course, these are all just guesstimates, for example for random complex-valued coefficients, none of these cancellations will have any effect at all.